src/frame/endpoints/UserApiEndpoints.ts

```diff
@@ -179,6 +179,26 @@ const userUpdate = (userID: string, userPrivateKey?: PrivateKey<Uint8Array>, sta
     errorCode: ErrorCodes.USER_UPDATE_REQUEST_FAILURE,
 });
 
+const userKeyUpdateApi = (
+    userID: string,
+    userPrivateKey: PrivateKey<Uint8Array>,
+    augmentationFactor: AugmentationFactor<Uint8Array>,
+    userKeyId: number
+): RequestMeta => ({
+    url: `users/${encodeURIComponent(userID)}keys/${userKeyId}`,
+    options: {
+        method: "PUT",
+        headers: {
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            userPrivateKey: userPrivateKey ? fromByteArray(userPrivateKey) : undefined,
+            augmentationFactor: augmentationFactor ? fromByteArray(augmentationFactor) : undefined,
```

Comment:
Can both of these really be optional?

Reply:
They might be optional for ironcore-id, but here both the `userPrivateKey` and `augmentationFactor` are marked as fixed fields, they aren't options. So there's no need to check whether they exist.